### PR TITLE
Make HFDatasetDataModule a datasets.load_dataset wrapper

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Before:
```
dataset = load_dataset("rajpurkar/squad", split="train")
datamodule = llm.HFDatasetDataModule(dataset, pad_token_id=tokenizer.eos_token_id)
```

After:
```
datamodule = llm.HFDatasetDataModule("rajpurkar/squad", split="train", pad_token_id=tokenizer.eos_token_id)
```

Some additional use-cases:

- Load multiple partitions:
```
datamodule = llm.HFDatasetDataModule("rajpurkar/squad", split=["train", "validation"], pad_token_id=tokenizer.eos_token_id)
```

- Custom collate-fn:
```
def my_collate_fn(examples):
    ....
datamodule = llm.HFDatasetDataModule("rajpurkar/squad", split=["train", "validation"], collate_fn=my_collate_fn)
```

- Custom map function:
```
datamodule = llm.HFDatasetDataModule("rajpurkar/squad", split="train", pad_token_id=tokenizer.eos_token_id)
datamodule.map(formatting_prompts_func, batched=False, batch_size=2)
```
Instead of calling `.map` on the dataset we call it on the datamodule (which forwards it to the dataset).

**Note:** We expect the user to take appropriate care so that the keys in the returned batch (dict of tensors) matches the arguments of the downstream model.

**Collection**: LLM

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by danielkorzekwa** [comment]
datamodule = llm.HFDatasetDataModule("rajpurkar/squad", split="train", pad_token_id=tokenizer.eos_token_id) - this solution enforces using a particular dataset split, 

looking into finetune_default:
```
 # datamodule = run.Config(llm.SquadDataModule, seq_length=2048, global_batch_size=128, micro_batch_size=1)
```

no split is provided and the the datamodule is able to provide both train and validation dataloaders so that later

```
recipe.trainer.val_check_interval = 100
```

can control whether validation should be active or not.

**Review by marcromeyn — APPROVED** [review]
